streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBStore.java

```diff
@@ -229,40 +230,34 @@ void openDB(final Map<String, Object> configs, final File stateDir) {
 
         // Setup statistics before the database is opened, otherwise the statistics are not updated
         // with the measurements from Rocks DB
-        maybeSetUpStatistics(configs);
-
+        statistics = userSpecifiedOptions.statistics();
+        if (statistics == null) {
+            if (RecordingLevel.forName((String) configs.get(METRICS_RECORDING_LEVEL_CONFIG)) == RecordingLevel.DEBUG) {
+                statistics = new Statistics();
+                dbOptions.setStatistics(statistics);
+            }
+            userSpecifiedStatistics = false;
+        } else {
+            userSpecifiedStatistics = true;
+        }
```

Comment:
Could you extract this to a method? IMO, `openDB()` is already quite long.

Reply:
For me personally, spreading linear code across methods with only one call-site does not help readability. Anyway, going with the PMC member on this one, done ;)

Reply:
While I get the joke, it is not a matter of PMC member or not. If you have good arguments, I am happy to be convinced.

When I read code, I do not want to be distracted by every detail right away. For the first overview when reading a method -- for me -- it is enough to know that the statistics are set up. So reading `setupStatistics()` would be enough for me. Then, if I need to understand how the statistics are setup I would look into setupStatistics().

Reply:
Yes, I was just joking. Normally, I'd be happy to convince you with arguments, but I think in this case it's really a matter of personal preference or convention among the engineers in the project (and you know those better, honestly!).

In this particular case, I think a comment and newlines help to split the function into correlated "blocks".

```
// setup statistics
...
...

// next block
```

Arguably, this is not as clearly scoped as a separate method, as you say. But it does give me extra information when reading: I do not at all have to worry about this code if I am trying to understand any of the other methods, because the code cannot be called from other methods. Also, I don't have to use my IDE to jump around to understand the flow of execution. 

Everyone will make these kinds of trade-offs slightly differently, I am happy to err on the side of project convention here.

Reply:
I am not a big fan of comments. They eventually start to lie because they are not maintained as well as executable parts of the code. I prefer to extract a code block into a method with a meaningful name instead of using a inline comment. One could argue, that the name of a method might also start to lie, but IMO that is less likely compared to an inline comment.

Reply:
yeah watch out @lucasbru, don't mention code comments around @cadonna or he'll come after you 😜

Reply:
No comment!